<a href="https://colab.research.google.com/github/ie-katerinai/Python_Analitik/blob/main/Seminar_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

Считать данные с помощью pandas

In [ ]:
df = pd.read_csv('./laptop_price.csv', encoding = '1251')

In [ ]:
df

laptop_ID Company                              Product  \
0             1   Apple                          MacBook Pro   
1             2   Apple                          Macbook Air   
2             3      HP                               250 G6   
3             4   Apple                          MacBook Pro   
4             5   Apple                          MacBook Pro   
...         ...     ...                                  ...   
1298       1316  Lenovo                       Yoga 500-14ISK   
1299       1317  Lenovo                       Yoga 900-13ISK   
1300       1318  Lenovo                   IdeaPad 100S-14IBR   
1301       1319      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon   
1302       1320    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)   

                TypeName  Inches                            ScreenResolution  \
0              Ultrabook    13.3          IPS Panel Retina Display 2560x1600   
1              Ultrabook    13.3                                    1440x900   
2               Notebook    15.6                           Full HD 1920x1080   
3              Ultrabook    15.4          IPS Panel Retina Display 2880x1800   
4              Ultrabook    13.3          IPS Panel Retina Display 2560x1600   
...                  ...     ...                                         ...   
1298  2 in 1 Convertible    14.0   IPS Panel Full HD / Touchscreen 1920x1080   
1299  2 in 1 Convertible    13.3  IPS Panel Quad HD+ / Touchscreen 3200x1800   
1300            Notebook    14.0                                    1366x768   
1301            Notebook    15.6                                    1366x768   
1302            Notebook    15.6                                    1366x768   

                                       Cpu   Ram               Memory  \
0                     Intel Core i5 2.3GHz   8GB            128GB SSD   
1                     Intel Core i5 1.8GHz   8GB  128GB Flash Storage   
2               Intel Core i5 7200U 2.5GHz   8GB            256GB SSD   
3                     Intel Core i7 2.7GHz  16GB            512GB SSD   
4                     Intel Core i5 3.1GHz   8GB            256GB SSD   
...                                    ...   ...                  ...   
1298            Intel Core i7 6500U 2.5GHz   4GB            128GB SSD   
1299            Intel Core i7 6500U 2.5GHz  16GB            512GB SSD   
1300  Intel Celeron Dual Core N3050 1.6GHz   2GB   64GB Flash Storage   
1301            Intel Core i7 6500U 2.5GHz   6GB              1TB HDD   
1302  Intel Celeron Dual Core N3050 1.6GHz   4GB            500GB HDD   

                               Gpu       OpSys  Weight  Price_euros  
0     Intel Iris Plus Graphics 640       macOS  1.37kg      1339.69  
1           Intel HD Graphics 6000       macOS  1.34kg       898.94  
2            Intel HD Graphics 620       No OS  1.86kg       575.00  
3               AMD Radeon Pro 455       macOS  1.83kg      2537.45  
4     Intel Iris Plus Graphics 650       macOS  1.37kg      1803.60  
...                            ...         ...     ...          ...  
1298         Intel HD Graphics 520  Windows 10   1.8kg       638.00  
1299         Intel HD Graphics 520  Windows 10   1.3kg      1499.00  
1300             Intel HD Graphics  Windows 10   1.5kg       229.00  
1301            AMD Radeon R5 M330  Windows 10  2.19kg       764.00  
1302             Intel HD Graphics  Windows 10   2.2kg       369.00  

[1303 rows x 13 columns]

Вывести на экран первые 5 строк


In [ ]:
df.head()

laptop_ID Company      Product   TypeName  Inches  \
0          1   Apple  MacBook Pro  Ultrabook    13.3   
1          2   Apple  Macbook Air  Ultrabook    13.3   
2          3      HP       250 G6   Notebook    15.6   
3          4   Apple  MacBook Pro  Ultrabook    15.4   
4          5   Apple  MacBook Pro  Ultrabook    13.3   

                     ScreenResolution                         Cpu   Ram  \
0  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                            1440x900        Intel Core i5 1.8GHz   8GB   
2                   Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3  IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4  IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   

                Memory                           Gpu  OpSys  Weight  \
0            128GB SSD  Intel Iris Plus Graphics 640  macOS  1.37kg   
1  128GB Flash Storage        Intel HD Graphics 6000  macOS  1.34kg   
2            256GB SSD         Intel HD Graphics 620  No OS  1.86kg   
3            512GB SSD            AMD Radeon Pro 455  macOS  1.83kg   
4            256GB SSD  Intel Iris Plus Graphics 650  macOS  1.37kg   

   Price_euros  
0      1339.69  
1       898.94  
2       575.00  
3      2537.45  
4      1803.60

Посмотреть на описание признаков и на их содержание

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         1303 non-null   int64  
 1   Company           1303 non-null   object 
 2   Product           1303 non-null   object 
 3   TypeName          1303 non-null   object 
 4   Inches            1303 non-null   float64
 5   ScreenResolution  1303 non-null   object 
 6   Cpu               1303 non-null   object 
 7   Ram               1303 non-null   object 
 8   Memory            1303 non-null   object 
 9   Gpu               1303 non-null   object 
 10  OpSys             1303 non-null   object 
 11  Weight            1303 non-null   object 
 12  Price_euros       1303 non-null   float64
dtypes: float64(2), int64(1), object(10)
memory usage: 132.5+ KB


Выведем основную статистику по указанным колонкам

In [ ]:
df[['Inches','Price_euros']].describe()

Inches  Price_euros
count  1303.000000  1303.000000
mean     15.017191  1123.686992
std       1.426304   699.009043
min      10.100000   174.000000
25%      14.000000   599.000000
50%      15.600000   977.000000
75%      15.600000  1487.880000
max      18.400000  6099.000000

вывести статистику по колонкам: 

-их общее число, 

-кол-во уникальных колонок, 

-часто встречающиеся ячейки, 

-частота 

In [ ]:
df.describe(include=['object'])

Company Product  TypeName   ScreenResolution  \
count     1303    1303      1303               1303   
unique      19     618         6                 40   
top       Dell  XPS 13  Notebook  Full HD 1920x1080   
freq       297      30       727                507   

                               Cpu   Ram     Memory                    Gpu  \
count                         1303  1303       1303                   1303   
unique                         118     9         39                    110   
top     Intel Core i5 7200U 2.5GHz   8GB  256GB SSD  Intel HD Graphics 620   
freq                           190   619        412                    281   

             OpSys Weight  
count         1303   1303  
unique           9    179  
top     Windows 10  2.2kg  
freq          1072    121

##Задача 2
Проведите первичный анализ данных.

Изучите типы данных.

Найдите количество пропущенных ячеек в данных.

Посчитайте основные статистики по всем признакам и поизучайте их.

Проверяем количество незаполненных ячеек по всем колонкам(таковых не нашлось)

In [ ]:
df.isnull()

laptop_ID  Company  Product  TypeName  Inches  ScreenResolution    Cpu  \
0         False    False    False     False   False             False  False   
1         False    False    False     False   False             False  False   
2         False    False    False     False   False             False  False   
3         False    False    False     False   False             False  False   
4         False    False    False     False   False             False  False   
...         ...      ...      ...       ...     ...               ...    ...   
1298      False    False    False     False   False             False  False   
1299      False    False    False     False   False             False  False   
1300      False    False    False     False   False             False  False   
1301      False    False    False     False   False             False  False   
1302      False    False    False     False   False             False  False   

        Ram  Memory    Gpu  OpSys  Weight  Price_euros  
0     False   False  False  False   False        False  
1     False   False  False  False   False        False  
2     False   False  False  False   False        False  
3     False   False  False  False   False        False  
4     False   False  False  False   False        False  
...     ...     ...    ...    ...     ...          ...  
1298  False   False  False  False   False        False  
1299  False   False  False  False   False        False  
1300  False   False  False  False   False        False  
1301  False   False  False  False   False        False  
1302  False   False  False  False   False        False  

[1303 rows x 13 columns]

In [ ]:
df.isnull().sum()

laptop_ID           0
Company             0
Product             0
TypeName            0
Inches              0
ScreenResolution    0
Cpu                 0
Ram                 0
Memory              0
Gpu                 0
OpSys               0
Weight              0
Price_euros         0
dtype: int64

То же, что и df.isnull()

In [ ]:
df.isna()

laptop_ID  Company  Product  TypeName  Inches  ScreenResolution    Cpu  \
0         False    False    False     False   False             False  False   
1         False    False    False     False   False             False  False   
2         False    False    False     False   False             False  False   
3         False    False    False     False   False             False  False   
4         False    False    False     False   False             False  False   
...         ...      ...      ...       ...     ...               ...    ...   
1298      False    False    False     False   False             False  False   
1299      False    False    False     False   False             False  False   
1300      False    False    False     False   False             False  False   
1301      False    False    False     False   False             False  False   
1302      False    False    False     False   False             False  False   

        Ram  Memory    Gpu  OpSys  Weight  Price_euros  
0     False   False  False  False   False        False  
1     False   False  False  False   False        False  
2     False   False  False  False   False        False  
3     False   False  False  False   False        False  
4     False   False  False  False   False        False  
...     ...     ...    ...    ...     ...          ...  
1298  False   False  False  False   False        False  
1299  False   False  False  False   False        False  
1300  False   False  False  False   False        False  
1301  False   False  False  False   False        False  
1302  False   False  False  False   False        False  

[1303 rows x 13 columns]

То же, что и df.isnull().sum()

In [ ]:
df.isna().sum()

laptop_ID           0
Company             0
Product             0
TypeName            0
Inches              0
ScreenResolution    0
Cpu                 0
Ram                 0
Memory              0
Gpu                 0
OpSys               0
Weight              0
Price_euros         0
dtype: int64

##Задача 3 
Ответьте на несколько вопросов 

**3.1** Ноутбуков от какой компании больше всего в наборе данных? 

Выведем количество всех ноутбуков по фирмам, чтобы проверить следующий результат. 

.value_counts() - показывает количество уникальных значений.

In [ ]:
df['Company'].value_counts()

Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Xiaomi         4
Vero           4
Chuwi          3
Google         3
Fujitsu        3
LG             3
Huawei         2
Name: Company, dtype: int64

Посмотреть фирмы и их продукты

In [ ]:
df[['Company','Product']]

Company                              Product
0      Apple                          MacBook Pro
1      Apple                          Macbook Air
2         HP                               250 G6
3      Apple                          MacBook Pro
4      Apple                          MacBook Pro
...      ...                                  ...
1298  Lenovo                       Yoga 500-14ISK
1299  Lenovo                       Yoga 900-13ISK
1300  Lenovo                   IdeaPad 100S-14IBR
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)

[1303 rows x 2 columns]

Вывод компаний и его продуктов(моделей) от максимального __количества__ к минимальному

In [ ]:
df[['Company','Product']].value_counts()

Company  Product                                 
Dell     XPS 13                                      30
         Inspiron 3567                               29
HP       250 G6                                      21
Lenovo   Legion Y520-15IKBN                          19
Dell     Vostro 3568                                 19
                                                     ..
HP       Elitebook 1040                               1
Asus     Rog G752VT-GC073T                            1
         Rog G752VY-GC229T                            1
         Rog GL502VM-DS74                             1
Acer     A715-71G-59DH (i5-7300HQ/8GB/1TB/GeForce     1
Length: 620, dtype: int64

Выведим максимальное количество ноутбуков и их фирму(Dell 297, Lenovo 297 - то, что должно получиться).

Вывелость только количество.

In [ ]:
df['Company'].value_counts().max()

297

Вывелась только фирма

.idxmax() - выводит индекс максимального значения

In [ ]:
df['Company'].value_counts().idxmax()

'Dell'

Чтобы вывести в нужной нам форме и два производителя, напишем лябда-функцию.

Функция loc выводит данные по индексам(первые несколько). 

Функция лямбда задаеткакие именно индексы нужно взять - с максимальным значением - и передает их в Х - это уже число. 

Loc[х] - выводит строки с данными

In [ ]:
df['Company'].value_counts().loc[lambda x: x==df['Company'].value_counts().max()]

Dell      297
Lenovo    297
Name: Company, dtype: int64

**3.2** Какая минимальная и максимальная стоимость у ноутбуков в данных?

In [ ]:
df['Price_euros'].min(), df['Price_euros'].max()

(174.0, 6099.0)

**3.3** Какой самый дорогой ноутбук в данных? 

Выведите все характеристики только по этому ноутбуку 

Если таких ноутбуков несколько, то выводите их всех

Вывели фирму самого дорогого ноутука

In [ ]:
df['Company'].value_counts().idxmax()

'Dell'

вывели все ноутбуки той фирмы, у которой самый дорогой ноутбук

In [ ]:
df[df['Company'] ==  df['Company'].value_counts().idxmax()]

laptop_ID Company        Product            TypeName  Inches  \
13           14    Dell  Inspiron 3567            Notebook    15.6   
16           17    Dell  Inspiron 3567            Notebook    15.6   
19           20    Dell         XPS 13           Ultrabook    13.3   
23           24    Dell  Inspiron 5379  2 in 1 Convertible    13.3   
25           26    Dell  Inspiron 3567            Notebook    15.6   
...         ...     ...            ...                 ...     ...   
1278       1296    Dell  Inspiron 3552            Notebook    15.6   
1281       1299    Dell  Inspiron 3567            Notebook    15.6   
1290       1308    Dell  Inspiron 3567            Notebook    15.6   
1292       1310    Dell  Inspiron 3552            Notebook    15.6   
1295       1313    Dell  Inspiron 3567            Notebook    15.6   

                               ScreenResolution  \
13                            Full HD 1920x1080   
16                            Full HD 1920x1080   
19    IPS Panel Full HD / Touchscreen 1920x1080   
23              Full HD / Touchscreen 1920x1080   
25                                     1366x768   
...                                         ...   
1278                                   1366x768   
1281                                   1366x768   
1290                                   1366x768   
1292                                   1366x768   
1295                                   1366x768   

                                       Cpu  Ram     Memory  \
13                Intel Core i3 6006U 2GHz  4GB  256GB SSD   
16              Intel Core i7 7500U 2.7GHz  8GB  256GB SSD   
19              Intel Core i5 8250U 1.6GHz  8GB  128GB SSD   
23              Intel Core i5 8250U 1.6GHz  8GB  256GB SSD   
25                Intel Core i3 6006U 2GHz  4GB    1TB HDD   
...                                    ...  ...        ...   
1278  Intel Celeron Dual Core N3050 1.6GHz  2GB  500GB HDD   
1281            Intel Core i7 7500U 2.7GHz  8GB    1TB HDD   
1290            Intel Core i3 7100U 2.4GHz  6GB    1TB HDD   
1292  Intel Celeron Dual Core N3050 1.6GHz  2GB  500GB HDD   
1295            Intel Core i7 7500U 2.7GHz  8GB    1TB HDD   

                         Gpu       OpSys  Weight  Price_euros  
13        AMD Radeon R5 M430  Windows 10   2.2kg       498.90  
16        AMD Radeon R5 M430  Windows 10   2.2kg       745.00  
19    Intel UHD Graphics 620  Windows 10  1.22kg       979.00  
23    Intel UHD Graphics 620  Windows 10  1.62kg       819.00  
25     Intel HD Graphics 520  Windows 10   2.3kg       418.64  
...                      ...         ...     ...          ...  
1278       Intel HD Graphics  Windows 10  2.20kg       379.00  
1281      AMD Radeon R5 M430       Linux   2.3kg       805.99  
1290   Intel HD Graphics 620  Windows 10   2.3kg       459.00  
1292       Intel HD Graphics  Windows 10  2.20kg       379.00  
1295      AMD Radeon R5 M430       Linux   2.3kg       805.99  

[297 rows x 13 columns]

Выводим данные по самому дорогому ноутбуку

In [ ]:
df[df['Price_euros'] ==  df['Price_euros'].max()]

laptop_ID Company    Product TypeName  Inches  \
196        200   Razer  Blade Pro   Gaming    17.3   

                        ScreenResolution                          Cpu   Ram  \
196  4K Ultra HD / Touchscreen 3840x2160  Intel Core i7 7820HK 2.9GHz  32GB   

      Memory                      Gpu       OpSys  Weight  Price_euros  
196  1TB SSD  Nvidia GeForce GTX 1080  Windows 10  3.49kg       6099.0

вывели все по Dell

In [ ]:
df[df['Company']==df['Company'].value_counts().idxmax()]

laptop_ID Company        Product            TypeName  Inches  \
13           14    Dell  Inspiron 3567            Notebook    15.6   
16           17    Dell  Inspiron 3567            Notebook    15.6   
19           20    Dell         XPS 13           Ultrabook    13.3   
23           24    Dell  Inspiron 5379  2 in 1 Convertible    13.3   
25           26    Dell  Inspiron 3567            Notebook    15.6   
...         ...     ...            ...                 ...     ...   
1278       1296    Dell  Inspiron 3552            Notebook    15.6   
1281       1299    Dell  Inspiron 3567            Notebook    15.6   
1290       1308    Dell  Inspiron 3567            Notebook    15.6   
1292       1310    Dell  Inspiron 3552            Notebook    15.6   
1295       1313    Dell  Inspiron 3567            Notebook    15.6   

                               ScreenResolution  \
13                            Full HD 1920x1080   
16                            Full HD 1920x1080   
19    IPS Panel Full HD / Touchscreen 1920x1080   
23              Full HD / Touchscreen 1920x1080   
25                                     1366x768   
...                                         ...   
1278                                   1366x768   
1281                                   1366x768   
1290                                   1366x768   
1292                                   1366x768   
1295                                   1366x768   

                                       Cpu  Ram     Memory  \
13                Intel Core i3 6006U 2GHz  4GB  256GB SSD   
16              Intel Core i7 7500U 2.7GHz  8GB  256GB SSD   
19              Intel Core i5 8250U 1.6GHz  8GB  128GB SSD   
23              Intel Core i5 8250U 1.6GHz  8GB  256GB SSD   
25                Intel Core i3 6006U 2GHz  4GB    1TB HDD   
...                                    ...  ...        ...   
1278  Intel Celeron Dual Core N3050 1.6GHz  2GB  500GB HDD   
1281            Intel Core i7 7500U 2.7GHz  8GB    1TB HDD   
1290            Intel Core i3 7100U 2.4GHz  6GB    1TB HDD   
1292  Intel Celeron Dual Core N3050 1.6GHz  2GB  500GB HDD   
1295            Intel Core i7 7500U 2.7GHz  8GB    1TB HDD   

                         Gpu       OpSys  Weight  Price_euros  
13        AMD Radeon R5 M430  Windows 10   2.2kg       498.90  
16        AMD Radeon R5 M430  Windows 10   2.2kg       745.00  
19    Intel UHD Graphics 620  Windows 10  1.22kg       979.00  
23    Intel UHD Graphics 620  Windows 10  1.62kg       819.00  
25     Intel HD Graphics 520  Windows 10   2.3kg       418.64  
...                      ...         ...     ...          ...  
1278       Intel HD Graphics  Windows 10  2.20kg       379.00  
1281      AMD Radeon R5 M430       Linux   2.3kg       805.99  
1290   Intel HD Graphics 620  Windows 10   2.3kg       459.00  
1292       Intel HD Graphics  Windows 10  2.20kg       379.00  
1295      AMD Radeon R5 M430       Linux   2.3kg       805.99  

[297 rows x 13 columns]

Выводим отсортированные значения по цене, видим, что самый дорогой только один.

In [ ]:
df['Price_euros'].value_counts().sort_index()

174.0     1
191.9     1
196.0     1
199.0     2
202.9     1
         ..
3975.0    1
4389.0    1
4899.0    1
5499.0    1
6099.0    1
Name: Price_euros, Length: 791, dtype: int64

In [ ]:
df[df['Price_euros']==6099.0]

laptop_ID Company    Product TypeName  Inches  \
196        200   Razer  Blade Pro   Gaming    17.3   

                        ScreenResolution                          Cpu   Ram  \
196  4K Ultra HD / Touchscreen 3840x2160  Intel Core i7 7820HK 2.9GHz  32GB   

      Memory                      Gpu       OpSys  Weight  Price_euros  
196  1TB SSD  Nvidia GeForce GTX 1080  Windows 10  3.49kg       6099.0

##Задача 4
Ответьте на несколько вопросов

**4.1** 

Найдите ноутбуки с самой маленькой диагональю в данных? 

Выведите только следующие характеристики:

-Компания

-Типа ноутбука

-Диагональ

-Стоимость

Если таких ноутбуков несколько, то выводите их всех



In [ ]:
df[df['Inches']==df['Inches'].min()]

laptop_ID Company    Product            TypeName  Inches  \
50           51  Lenovo  Yoga Book  2 in 1 Convertible    10.1   
1082       1097  Lenovo  Yoga Book  2 in 1 Convertible    10.1   
1114       1129  Lenovo  Yoga Book  2 in 1 Convertible    10.1   
1126       1141  Lenovo  Yoga Book  2 in 1 Convertible    10.1   

                     ScreenResolution                          Cpu  Ram  \
50    IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   
1082  IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   
1114  IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   
1126  IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   

                  Memory                    Gpu       OpSys  Weight  \
50    64GB Flash Storage  Intel HD Graphics 400     Android  0.69kg   
1082  64GB Flash Storage  Intel HD Graphics 400  Windows 10  0.69kg   
1114  64GB Flash Storage  Intel HD Graphics 400     Android  0.69kg   
1126  64GB Flash Storage  Intel HD Graphics 400  Windows 10  0.69kg   

      Price_euros  
50         319.00  
1082       646.27  
1114       549.00  
1126       479.00

In [ ]:
df[df['Inches']==df['Inches'].min()][['Company','TypeName','Inches','Price_euros']]

Company            TypeName  Inches  Price_euros
50    Lenovo  2 in 1 Convertible    10.1       319.00
1082  Lenovo  2 in 1 Convertible    10.1       646.27
1114  Lenovo  2 in 1 Convertible    10.1       549.00
1126  Lenovo  2 in 1 Convertible    10.1       479.00

**4.2** Сколько стоит самый дорогой ноутбук у компании HP?




In [ ]:
df[df['Company']=='HP']['Price_euros'].max()

4389.0

**4.3** 

Как много ноутбуков Ultrabook с 8GB RAM? 

Найдите сколько таких ультрабуков с 8GB ОЗУ в процентном соотношении относительно всех ультрабуков.

In [ ]:
df[(df['TypeName']=='Ultrabook') & (df['Ram']=='8GB')].value_counts()

laptop_ID  Company  Product               TypeName   Inches  ScreenResolution                     Cpu                         Ram  Memory     Gpu                           OpSys       Weight  Price_euros
1          Apple    MacBook Pro           Ultrabook  13.3    IPS Panel Retina Display 2560x1600   Intel Core i5 2.3GHz        8GB  128GB SSD  Intel Iris Plus Graphics 640  macOS       1.37kg  1339.69        1
924        HP       Elitebook Folio       Ultrabook  12.5    4K Ultra HD / Touchscreen 3840x2160  Intel Core M 6Y75 1.2GHz    8GB  240GB SSD  Intel HD Graphics 515         Windows 10  1.09kg  3100.00        1
886        Dell     Latitude 7280         Ultrabook  12.5    Full HD 1920x1080                    Intel Core i7 7600U 2.8GHz  8GB  256GB SSD  Intel HD Graphics 620         Windows 10  1.6kg   1690.00        1
888        Xiaomi   Mi Notebook           Ultrabook  13.3    IPS Panel Full HD 1920x1080          Intel Core i5 6200U 2.3GHz  8GB  256GB SSD  Nvidia GeForce 940MX       

Найдите сколько таких ультрабуков с 8GB ОЗУ в процентном соотношении относительно всех ультрабуков.

In [ ]:
ultrabook8GB = df[(df['TypeName']=='Ultrabook') & (df['Ram']=='8GB')].shape[0]
ultrabook = df[df['TypeName']=='Ultrabook'].shape[0]
result = round(ultrabook8GB/ultrabook*100,2)
result

71.43

Или короче то же самое, но в долях

In [ ]:
df[df['TypeName'] == 'Ultrabook']['Ram'].value_counts(normalize=True).loc['8GB']

0.7142857142857143

##Задача 5
**5.1** Выберите ноутбук клиенту Клиент хочет подобрать ноутбук с 8GB или 16GB ОЗУ на Windows 10 в стоимости до 500 евро, сколько у него вариантов?

In [ ]:
df[((df['Ram']=='8GB') | (df['Ram']=='16GB')) & (df['OpSys']=='Windows 10') & (df['Price_euros']<=500)]

laptop_ID Company                                   Product  TypeName  \
74           76    Acer                             Aspire E5-475  Notebook   
175         179    Dell                             Inspiron 3567  Notebook   
256         261      HP        15-BW094nd (A6-9220/8GB/128GB/W10)  Notebook   
292         297      HP         17-BS037cl (i3-6006U/8GB/1TB/W10)  Notebook   
313         318    Acer  ES1-523-84K7 (A8-7410/8GB/256GB/FHD/W10)  Notebook   
365         371      HP            15-AY023na (N3710/8GB/2TB/W10)  Notebook   
571         578      HP            14-am079na (N3710/8GB/2TB/W10)  Notebook   
595         602  Lenovo                         IdeaPad 110-15ISK  Notebook   
634         642    Asus         X541NA-GO414T (N3350/8GB/1TB/W10)  Notebook   
702         710  Lenovo                         IdeaPad 320-15ABR  Notebook   
1057       1071    Acer                            Aspire ES1-523  Notebook   
1163       1181  Lenovo                         IdeaPad 310-15ISK  Notebook   
1232       1250    Dell                             Inspiron 3567  Notebook   

      Inches      ScreenResolution                                   Cpu  Ram  \
74      14.0              1366x768              Intel Core i3 6006U 2GHz  8GB   
175     15.6              1366x768            Intel Core i3 7100U 2.4GHz  8GB   
256     15.6              1366x768          AMD A6-Series A6-9220 2.5GHz  8GB   
292     17.3              1600x900              Intel Core i3 6006U 2GHz  8GB   
313     15.6     Full HD 1920x1080             AMD A8-Series 7410 2.2GHz  8GB   
365     15.6              1366x768  Intel Pentium Quad Core N3710 1.6GHz  8GB   
571     14.0              1366x768  Intel Pentium Quad Core N3710 1.6GHz  8GB   
595     15.6              1366x768            Intel Core i3 6100U 2.3GHz  8GB   
634     15.6              1366x768  Intel Celeron Dual Core N3350 1.1GHz  8GB   
702     15.6              1366x768           AMD A12-Series 9720P 3.6GHz  8GB   
1057    15.6              1366x768             AMD A8-Series 7410 2.2GHz  8GB   
1163    15.6              1366x768            Intel Core i3 6006U 2.0GHz  8GB   
1232    15.6  Touchscreen 1366x768            Intel Core i3 7100U 2.4GHz  8GB   

         Memory                     Gpu       OpSys  Weight  Price_euros  
74      1TB HDD   Intel HD Graphics 520  Windows 10   2.1kg       389.00  
175     1TB HDD   Intel HD Graphics 620  Windows 10   2.3kg       459.00  
256   128GB SSD  AMD Radeon R4 Graphics  Windows 10  1.91kg       445.90  
292     1TB HDD   Intel HD Graphics 520  Windows 10  2.54kg       489.00  
313   256GB SSD           AMD Radeon R5  Windows 10  2.23kg       469.00  
365     2TB HDD   Intel HD Graphics 405  Windows 10  2.04kg       389.00  
571     2TB HDD   Intel HD Graphics 405  Windows 10  1.94kg       389.00  
595     1TB HDD   Intel HD Graphics 520  Windows 10  2.29kg       459.00  
634     1TB HDD   Intel HD Graphics 500  Windows 10     2kg       399.00  
702     1TB HDD           AMD Radeon R7  Windows 10   2.2kg       429.00  
1057    1TB HDD           AMD Radeon R5  Windows 10   2.4kg       449.00  
1163    1TB HDD   Intel HD Graphics 520  Windows 10   2.2kg       459.00  
1232    1TB HDD   Intel HD Graphics 620  Windows 10   2.3kg       469.01

In [ ]:
df[((df['Ram']=='8GB') | (df['Ram']=='16GB')) & (df['OpSys']=='Windows 10') & (df['Price_euros']<=500)].shape[0]

13

**5.2** Выберите ноутбук клиенту 

Клиент хочет подобрать ноутбук от MSI, с видеокартой Nvidia GeForce GTX 1050 Ti и главное не с диагональю 15.6. 

В какой ценовой категории вышли подобные ноутбуки?

In [ ]:
nout=df[(df['Company']=='MSI') & (df['Gpu']=='Nvidia GeForce GTX 1050 Ti') & (df['Inches']!=15.6)]
nout

laptop_ID Company        Product TypeName  Inches   ScreenResolution  \
330         336     MSI     GL72M 7REX   Gaming    17.3  Full HD 1920x1080   
393         400     MSI  Leopard GP72M   Gaming    17.3  Full HD 1920x1080   
724         732     MSI     GL72M 7REX   Gaming    17.3  Full HD 1920x1080   
917         931     MSI     GE72VR 6RF   Gaming    17.3  Full HD 1920x1080   
1156       1174     MSI     GP72M 7REX   Gaming    17.3  Full HD 1920x1080   

                              Cpu   Ram                Memory  \
330   Intel Core i7 7700HQ 2.8GHz   8GB  256GB SSD +  1TB HDD   
393   Intel Core i7 7700HQ 2.8GHz   8GB             256GB SSD   
724   Intel Core i7 7700HQ 2.8GHz   8GB  128GB SSD +  1TB HDD   
917   Intel Core i7 7700HQ 2.8GHz  16GB  256GB SSD +  1TB HDD   
1156  Intel Core i7 7700HQ 2.8GHz  16GB  256GB SSD +  1TB HDD   

                             Gpu       OpSys Weight  Price_euros  
330   Nvidia GeForce GTX 1050 Ti  Windows 10  2.7kg      1199.00  
393   Nvidia GeForce GTX 1050 Ti  Windows 10  2.7kg      1349.00  
724   Nvidia GeForce GTX 1050 Ti  Windows 10  2.7kg      1348.48  
917   Nvidia GeForce GTX 1050 Ti  Windows 10  2.7kg      1599.00  
1156  Nvidia GeForce GTX 1050 Ti  Windows 10  2.7kg      1492.80

In [ ]:
new_s = nout['Price_euros'].describe()
new_s

count       5.000000
mean     1397.656000
std       153.167946
min      1199.000000
25%      1348.480000
50%      1349.000000
75%      1492.800000
max      1599.000000
Name: Price_euros, dtype: float64

В какой ценовой категории вышли подобные ноутбуки?

In [ ]:
new_sp = df[(df['Company']=='MSI') & (df['Gpu']=='Nvidia GeForce GTX 1050 Ti') & (df['Inches']!=15.6)]['Price_euros']
new_sp.min(), new_s.max()

(1199.0, 1599.0)

**5.3** 

Что дешевле? 

В среднем дешевле ноутбуки с CPU Intel Core i7 7700HQ 2.8GHz или с Intel Core i7 7600U 2.8GHz?

вывод всех ноутбуков с заданными Cpu

In [ ]:
df[df['Cpu'].isin(['Intel Core i7 7700HQ 2.8GHz','Intel Core i7 7600U 2.8GHz'])]

laptop_ID Company                                    Product   TypeName  \
41           42    Dell                              Inspiron 7577     Gaming   
58           59     MSI                                 GS73VR 7RG     Gaming   
65           67    Asus                                ZenBook Pro  Ultrabook   
69           71    Asus  FX753VE-GC093 (i7-7700HQ/12GB/1TB/GeForce     Gaming   
75           77    Asus  FX503VD-E4022T (i7-7700HQ/8GB/1TB/GeForce     Gaming   
...         ...     ...                                        ...        ...   
1214       1232    Asus                                  ROG Strix     Gaming   
1229       1247     MSI                                 GL62M 7RDX     Gaming   
1231       1249   Razer                                  Blade Pro     Gaming   
1233       1251     MSI                              GE72VR Apache     Gaming   
1245       1263    Asus                         Rog GL553VE-FY052T     Gaming   

      Inches             ScreenResolution                          Cpu   Ram  \
41      15.6  IPS Panel Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   
58      17.3            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   
65      15.6            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   
69      17.3            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  12GB   
75      15.6            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz   8GB   
...      ...                          ...                          ...   ...   
1214    17.3            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   
1229    15.6            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz   8GB   
1231    14.0            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   
1233    17.3            Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   
1245    15.6  IPS Panel Full HD 1920x1080  Intel Core i7 7700HQ 2.8GHz  16GB   

                    Memory                         Gpu       OpSys  Weight  \
41    256GB SSD +  1TB HDD     Nvidia GeForce GTX 1060  Windows 10  2.65kg   
58    256GB SSD +  2TB HDD     Nvidia GeForce GTX 1070  Windows 10  2.43kg   
65               512GB SSD  Nvidia GeForce GTX 1050 Ti  Windows 10   1.8kg   
69                 1TB HDD  Nvidia GeForce GTX 1050 Ti       Linux     3kg   
75                 1TB HDD     Nvidia GeForce GTX 1050  Windows 10   2.2kg   
...                    ...                         ...         ...     ...   
1214  256GB SSD +  1TB HDD     Nvidia GeForce GTX 1060  Windows 10   2.9kg   
1229  128GB SSD +  1TB HDD     Nvidia GeForce GTX 1050  Windows 10   2.2kg   
1231               1TB SSD     Nvidia GeForce GTX 1060  Windows 10  1.95kg   
1233  256GB SSD +  1TB HDD     Nvidia GeForce GTX 1060  Windows 10   2.7kg   
1245  256GB SSD +  1TB HDD  Nvidia GeForce GTX 1050 Ti  Windows 10   2.5kg   

      Price_euros  
41         1499.0  
58         2449.0  
65         1983.0  
69          949.0  
75          949.0  
...           ...  
1214       1949.0  
1229       1149.0  
1231       3499.0  
1233       1598.0  
1245       1600.0  

[159 rows x 13 columns]

In [ ]:
df[df['Cpu']=='Intel Core i7 7700HQ 2.8GHz']['Price_euros'].mean() < df[df['Cpu']=='Intel Core i7 7600U 2.8GHz']['Price_euros'].mean()

True

Вывод: дешевле ноутбук с Intel Core i7 7700HQ 2.8GHz

##Задача 6

Найдите самый легкий ноутбук Но обратите внимание на тип и представление данных в признаке Weight, если что, замените в строке 'kg' на пустую строку через метод .str.replace()

In [ ]:
df['Weight']=df['Weight'].str.replace('kg', '').astype('float')

In [ ]:
df['Weight']

0       1.37
1       1.34
2       1.86
3       1.83
4       1.37
        ... 
1298    1.80
1299    1.30
1300    1.50
1301    2.19
1302    2.20
Name: Weight, Length: 1303, dtype: float64

In [ ]:
df[df['Weight']==df['Weight'].min()]['Product']

50      Yoga Book
1082    Yoga Book
1114    Yoga Book
1126    Yoga Book
Name: Product, dtype: object